examples/tf-native/vsphere/hcl/folder.tf

```diff
@@ -0,0 +1,70 @@
+############################
+# Deploying vSphere Folder #
+############################
+
+provider "vsphere" {
+  user           = var.vsphere-user
+  password       = var.vsphere-password
+  vsphere_server = var.vsphere-server
+
+  # If you have a self-signed cert
+  allow_unverified_ssl = var.vsphere-unverified-ssl
+}
+
+#############
+# Variables #
+#############
+
+variable "vsphere-user" {
+  type        = string
+  description = "VMware vSphere user name"
+}
+
+variable "vsphere-password" {
+  type        = string
+  description = "VMware vSphere password"
+}
+
+variable "vsphere-server" {
+  type        = string
+  description = "VMware vCenter server FQDN / IP"
+}
+
+variable "vsphere-unverified-ssl" {
+  type        = bool
+  description = "Is the VMware vCenter using a self signed certificated (true/false)"
+  default     = true
+}
+
+variable "vsphere-datacenter" {
+  type        = string
+  description = "VMware vSphere datacenter"
+}
+
+variable "folder-name" {
+  type        = string
+  description = "The name of folder"
+}
+
+variable "folder-type" {
+  type        = string
+  description = "The type of folder"
+}
+
```

Comment:
Great. This sample is simple and all variables have more readable properties, like `description` and `type`.